src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -452,18 +453,46 @@ void MinMdnsResolver::AdvancePendingResolverStates()
         else if (resolver->IsActiveOperationalParse())
         {
             MATTER_TRACE_SCOPE("Active operational delegate call", "MinMdnsResolver");
-            ResolvedNodeData nodeData;
+            ResolvedNodeData nodeResolvedData;
+            CHIP_ERROR err = CHIP_ERROR_UNINITIALIZED;
+
+            err = resolver->Take(nodeResolvedData);
+
+            if (mActiveResolves.HasBrowseFor(chip::Dnssd::DiscoveryType::kOperational))
+            {
+                DiscoveredNodeData nodeData;
+                if (err == CHIP_NO_ERROR)
+                {
+                    OperationalNodeBrowseData opNodeData;
+                    opNodeData.peerId     = nodeResolvedData.operationalData.peerId;
+                    opNodeData.hasZeroTTL = nodeResolvedData.operationalData.hasZeroTTL;
+                    nodeData.Set<OperationalNodeBrowseData>(opNodeData);
+                }
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                }
+                else if (mDiscoveryContext != nullptr)
+                {
+                    mDiscoveryContext->OnNodeDiscovered(nodeData);
+                }
+                else
+                {
+#if CHIP_MINMDNS_HIGH_VERBOSITY
+                    ChipLogError(Discovery, "No delegate to report operational node discovery");
+#endif
+                }
+            }
 
-            CHIP_ERROR err = resolver->Take(nodeData);
             if (err != CHIP_NO_ERROR)
             {
-                ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                ChipLogError(Discovery, "Failed to take ResolvedNodeData - result: %" CHIP_ERROR_FORMAT, err.Format());
             }
 
-            mActiveResolves.Complete(nodeData.operationalData.peerId);
+            mActiveResolves.Complete(nodeResolvedData.operationalData.peerId);
```

Comment:
It's really weird to be doing this if err is not success, but it predates your PR.  Please file a followup issue on this use of `nodeResolvedData` when it might contain arbitrary garbage.